oxidizer/src/db/connections.rs

```diff
@@ -0,0 +1,147 @@
+use async_trait::async_trait;
+use tokio_postgres::{Client, Config, NoTls};
+
+use crate::Error;
+
+#[async_trait]
+pub trait ConnectionProvider: Send + Sync + 'static {
+    async fn connect(&self) -> Result<Client, tokio_postgres::Error>;
+}
+
+pub async fn create_connection_provider(
+    config: Config,
+    ca_file: Option<&str>,
+) -> Result<Box<dyn ConnectionProvider>, Error> {
+    let prov: Box<dyn ConnectionProvider> = if let Some(ca_file) = ca_file {
+        cfg_if::cfg_if! {
+            if #[cfg(feature = "tls-rustls")] {
+                tls_rustls::create_rustls_provider(config, ca_file).await?
+            } else if #[cfg(feature = "tls-openssl")] {
+                tls_openssl::create_openssl_provider(config, ca_file).await?
+            } else {
```

Comment:
reverted the if here ... in case rustls is configured, it should be used even if someone forgot to disable default features